### PR TITLE
Expose bit resolution of ADC, PWM (and others as available)

### PR DESCRIPTION
**Pull request by rwaldron**
Simplified mechanism for exposing resolution values from capabilities response. Intended to supersede #144

Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

File: lib/firmata.js [hunk 1]
```diff
@@ -505,6 +523,12 @@ function Board(port, options, callback) {
     RES_TX3: 0x07,
   };
 
+  this.RESOLUTION = {
+    ADC: null,
```
Comment (soundanalogous) [inline lib/firmata.js]: What happens if the user skips the configurationQuery? Do we need defaults values set here instead?

Comment (soundanalogous) [inline lib/firmata.js]: I guess another way to think about this is do the default values live in firmata.js or in johnny-five (where the normalization of analog and pwm values takes place).

Comment (soundanalogous) [inline lib/firmata.js]: This is the only place in firmata.js where there is a reference to the resolution: https://github.com/firmata/firmata.js/blob/master/lib/firmata.js#L740. If we set defaults in firmata.js that param description could change to something like:

"The data to write to the pin between 0 and this.RESOLUTION.PWM."

Comment (rwaldron) [inline lib/firmata.js]: > What happens if the user skips the configurationQuery? Do we need defaults values set here instead?

That's reasonable, we should just use Arduino defaults. (update shortly)


> If we set defaults in firmata.js that param description could change to something like:
> "The data to write to the pin between 0 and this.RESOLUTION.PWM."

Sounds good

Comment (monteslu) [inline lib/firmata.js]: I like this better than my stab at it :)

👍

**Comment by soundanalogous** [comment]
LGTM

**Comment by rwaldron** [comment]
@soundanalogous @monteslu thanks for reviewing

**Pull request merged**